webapp/src/components/github_repo_selector.jsx

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import debounce from 'debounce-promise';
+import AsyncSelect from 'react-select/async';
+
+import {getStyleForReactSelect} from 'utils/styles';
+
+import Client from 'client';
+
+const searchDebounceDelay = 400;
+
+export default class GithubRepoSelector extends PureComponent {
+    static propTypes = {
+        required: PropTypes.bool,
+        theme: PropTypes.object.isRequired,
+        onChange: PropTypes.func.isRequired,
+        error: PropTypes.string,
```

Comment:
This component should handle only any errors related to the Github repo selection and should not receive additional errors. Those should be handled by the parent component.

Reply:
I haven't had time to do, sorry about that. Can you help here? Thanks

Reply:
@jfrerich is this issue resolved?

Reply:
Great catch @maria.  In the new commit I've addressed errors at the modal level by adding them inline with the submit button since that is what caused the error.

![image](https://user-images.githubusercontent.com/7575921/76362202-9765ef80-62ee-11ea-8ce6-a7cff1f4267d.png)

Reply:
PR to address this issue is found here.  https://github.com/dhadiseputro/mattermost-plugin-github/pull/3